diag_manager/fms_diag_yaml.F90

```diff
@@ -42,9 +40,112 @@ module fms_diag_yaml_mod
+                                              !! acceptable values are "latlon" and "index"
+  real, allocatable :: lat_lon_sub_region (:) !< Array that stores the grid point bounds for the sub region
+                                              !! to use if grid_type is set to "latlon"
+                                              !! [dim1_begin, dim1_end, dim2_begin, dim2_end,
+                                              !!  dim3_begin, dim3_end, dim4_begin, dim4_end]
+  integer, allocatable :: index_sub_region (:) !< Array that stores the index bounds for the sub region to
+                                               !! to use if grid_type is set to "index"
+                                               !! [dim1_begin, dim1_end, dim2_begin, dim2_end,
+                                               !!  dim3_begin, dim3_end, dim4_begin, dim4_end]
+  integer :: tile !< Tile number of the sub region, required if using the "index" grid type
+
+end type subRegion_type
+
+!> @brief type to hold the diag_file information
+type diagYamlFiles_type
+  character (len=:), private, allocatable :: file_fname !< file name
+  character (len=:), private, allocatable :: file_frequnit !< the frequency unit
+  integer, private    :: file_freq !< the frequency of data
+  character (len=:), private, allocatable :: file_timeunit !< The unit of time
+  character (len=:), private, allocatable :: file_unlimdim !< The name of the unlimited dimension
+  logical, private :: file_write !< false if the user doesn't want to the file to be created
+  character (len=:), private, allocatable :: string_file_write !< false if the user doesn’t want the file to be
+                                                               !! created (default is true).
+  character (len=:), private, allocatable :: file_realm !< The modeling realm that the variables come from
+  type(subRegion_type), private :: file_sub_region !< type containing info about the subregion, if any
+  integer, private :: file_new_file_freq !< Frequency for closing the existing file
+  character (len=:), private, allocatable :: file_new_file_freq_units !< Time units for creating a new file.
+                                                                      !! Required if “new_file_freq” used
+  character (len=:), private, allocatable :: file_start_time !< Time to start the file for the first time. Requires
+                                                             !! “new_file_freq”
+  integer, private :: file_duration !< How long the file should receive data after start time
+                                    !! in “file_duration_units”.  This optional field can only
+                                    !! be used if the start_time field is present.  If this field
+                                    !! is absent, then the file duration will be equal to the
+                                    !! frequency for creating new files.
```

Comment:
What does this really mean: "If this field is absent, then the <file duration> will be equal to the frequency for creating new files".  In particular, what does <file duration> mean. Is it a different field than file_duration. Seems like somewhere (maybe not necessarily in this module) there should be another function to encasulate the functionality. Like maybe caculate_actual_file_duration( diagYmlFile_type.file_duration, ... )  ?

Reply:
If the file_duration/file_duration_units is present (e.g 6 days) , diag_manager will write a new file every 6th day from the start time with data at whatever the new_file_frequency is. (e.g hourly data or whatever)

If the file_duration/file_duration_units is not present, diag_manager will write a new file at whatever the `new_file_frequency` is. 

This will probably change, but we are not there yet.

Reply:
I think that if a comments  switches from a description about the current class to something  that happens in  another class, then the comments should make that switch obvious. Perhaps change the comment to "If this field
is absent, then the diag_manager will use a file duration that will be equal to the frequency for creating new files." ?